io-engine/src/lvm/vg_pool.rs

```diff
@@ -0,0 +1,370 @@
+            .filter(|vg| vg.matches(opts))
+            .collect::<Vec<_>>();
+
+        Ok(vgs)
+    }
+
+    /// Import a volume group with the name provided or create one with the name
+    /// and disks provided currently only import is supported.
+    pub(crate) async fn import_or_create(
+        args: PoolArgs,
+    ) -> Result<VolumeGroup, Error> {
+        let vg = Self::import_inner(args).await?;
+        info!("The lvm vg pool '{}' has been created", vg.name());
+        Ok(vg)
+    }
+
+    async fn import_lvols(&self) -> Result<(), Error> {
+        self.list_lvs().await?;
+        Ok(())
+    }
+    async fn list_lvs(&self) -> Result<Vec<LogicalVolume>, Error> {
+        let query = super::QueryArgs::new()
+            .with_lv(CmnQueryArgs::ours())
+            .with_vg(CmnQueryArgs::ours().uuid(self.uuid()).named(self.name()));
+        LogicalVolume::list(&query).await
+    }
+
+    /// Import a volume group by its name, match the disks on the volume group
+    /// and if true add our tag to the volume group to make it available
+    /// as a Pool.
+    pub(crate) async fn import(args: PoolArgs) -> Result<VolumeGroup, Error> {
+        let vg = Self::import_inner(args).await?;
+        vg.import_lvols().await?;
+        Ok(vg)
+    }
+
+    /// Import a volume group by its name, match the disks on the volume group
+    /// and if true add our tag to the volume group to make it available
+    /// as a Pool.
+    async fn import_inner(args: PoolArgs) -> Result<VolumeGroup, Error> {
+        let name = &args.name;
+        let mut vg = Self::lookup(CmnQueryArgs::any().named(name)).await?;
+
+        if args.uuid.is_some() {
+            return Err(Error::VgUuidSet {});
+        }
+
+        if vg.disks != args.disks {
+            return Err(Error::DisksMismatch {
+                args: args.disks,
+                vg: vg.disks,
+            });
+        }
+
+        if vg.tags.contains(&Property::Lvm) {
+            return Ok(vg);
+        }
+
+        LvmCmd::vg_change(name)
+            .arg("-q")
+            .tag(Property::Lvm)
+            .run()
+            .await?;
+        vg.tags.push(Property::Lvm);
+        Ok(vg)
+    }
+
+    /// Delete the volume group.
+    /// > Note: Currently the vg is only exported.
+    /// todo: currently only exporting the VG.
+    pub(crate) async fn destroy(mut self) -> Result<(), Error> {
+        // As currently only import of volume group is supported
+        // exporting the volume group on destroy.
+        let name = self.name().to_string();
+        self.export().await?;
+        info!("LVM pool '{name}' has been destroyed successfully");
+        self.ptpl().destroy().ok();
+        Ok(())
+    }
+
+    /// Exports the volume group by unloading all logical volumes and finally
+    /// removing our tag from it
+    pub(crate) async fn export(&mut self) -> Result<(), Error> {
+        // todo: unload all logical volumes from this VG
```

Comment:
We cannot delete a pool having replicas right, so should we fail? And even if we want to destroy in use pool, should we not delete the replicas(spdk bdevs) as we created them instead of export?

Since we are using LVM as pool backend so I believe we don't need to deal with LVM LVs right?

Reply:
The LVM VG is a new Backend disk media ownership + management subsystem. SPDK and Blobstor are not involved anywhere in the disk management architecture. The user are making an explicit decision on the type of pool and the functionality they are looking for.
- LVM = LVM exclusively owns + manages the backend disk media & provides RAID 0, 1, 4, 5, 6, and 10 with full volume aggregation but average performance
- LVS = SPDK exclusively owns + manages the backend disk media but provides no RAID (yet) or disk aggregation (yet), but delivers very fast deep NVMe integration
- LZP = ZFS Zpool exclusively owns + manages the backend disk media & provides RAID 0, 1, Z (5), Z2 (6), Z3, and 10 with full volume aggregation and superior Bit-rot protection, Snapshot send/recv/streaming, Dedupe, Compression, and Flash optimizations (ARC and L2ARC).